Marlin/src/module/planner.cpp

```diff
@@ -2442,23 +2472,28 @@ bool Planner::_populate_block(
   #endif
 
   #if ENABLED(LIN_ADVANCE)
-    block->la_advance_rate = 0;
-    block->la_scaling = 0;
-
-    if (use_advance_lead) {
-      // the Bresenham algorithm will convert this step rate into extruder steps
-      block->la_advance_rate = extruder_advance_K[E_INDEX_N(extruder)] * block->acceleration_steps_per_s2;
+    #if ENABLED(SMOOTH_LIN_ADV)
+      block->use_advance_lead = use_advance_lead;
+      block->xy_to_e_step_ratio = (block->direction_bits.e ? 1 : -1) * 
```

Comment:
I found this name confusing. Partly because I read it backwards, i.e. xy to e ration would be xy / e in my head. Partly because xy may well not be x and/or y at all. Could even be e in fact. Maybe just `e_step_ratio` would be good.